pkg/resourcemanager/psql/psqluser/psqluser.go

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package psqluser
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"reflect"
+	"strings"
+
+	psql "github.com/Azure/azure-sdk-for-go/services/postgresql/mgmt/2017-12-01/postgresql"
+	"github.com/Azure/azure-service-operator/pkg/helpers"
+	"github.com/Azure/azure-service-operator/pkg/resourcemanager/config"
+	psdatabase "github.com/Azure/azure-service-operator/pkg/resourcemanager/psql/database"
+	"github.com/Azure/azure-service-operator/pkg/secrets"
+
+	"github.com/Azure/azure-service-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	_ "github.com/lib/pq" //the pg lib
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// PSqlServerPort is the default server port for sql server
+const PSqlServerPort = 5432
+
+// PDriverName is driver name for psqldb connection
+const PDriverName = "postgres"
+
+// PSecretUsernameKey is the username key in secret
+const PSecretUsernameKey = "username"
+
+// PSecretPasswordKey is the password key in secret
+const PSecretPasswordKey = "password"
+
+//PostgreSqlUserManager for psqluser manager
+type PostgreSqlUserManager struct {
+	SecretClient secrets.SecretClient
+	Scheme       *runtime.Scheme
+}
+
+//NewPostgreSqlUserManager creates a new PostgreSqlUserManager
+func NewPostgreSqlUserManager(secretClient secrets.SecretClient, scheme *runtime.Scheme) *PostgreSqlUserManager {
+	return &PostgreSqlUserManager{
+		SecretClient: secretClient,
+		Scheme:       scheme,
+	}
+}
+
+// GetDB retrieves a database
+func (s *PostgreSqlUserManager) GetDB(ctx context.Context, resourceGroupName string, serverName string, databaseName string) (db psql.Database, err error) {
+	dbClient, err := psdatabase.GetPSQLDatabasesClient()
+	if err != nil {
+		return psql.Database{}, err
+	}
+
+	return dbClient.Get(
+		ctx,
+		resourceGroupName,
+		serverName,
+		databaseName,
+	)
+}
+
+// ConnectToSqlDb connects to the PostgreSQL db using the given credentials
+func (s *PostgreSqlUserManager) ConnectToSqlDb(ctx context.Context, drivername string, server string, database string, port int, user string, password string) (*sql.DB, error) {
+
+	psqldbdnssuffix := "database.azure.com"
+	if config.Environment().Name != "AzurePublicCloud" {
+		psqldbdnssuffix = config.Environment().SQLDatabaseDNSSuffix
+	}
+	//the host or fullserveraddress should be:
+	//for public cloud <server>.postgres.database.azure.com
+	//for China cloud <server>.postgres.database.chinacloudapi.cn
+	//for German cloud <server>.postgres.database.cloudapi.de
+	//for US government <server>.postgres.database.usgovcloudapi.net
+	fullServerAddress := fmt.Sprintf("%s.postgres."+psqldbdnssuffix, server)
```

Comment:
Is this something we should be setting in the psql secret? I believe we have this for Azure SQL.

Reply:
This is already in the secret I guess

```
  "fullyQualifiedServerName": "your_server.postgres.database.azure.com",
```